### PR TITLE
Feature/show server events

### PR DESCRIPTION
**Pull request by Crackhoff**

### CODE REVIEW

File: compile_proto.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+git submodule update --recursive --remote --init >> /dev/null
+printf "DONE\n\n"
+
+file="./protos/jellyfish/server_notifications.proto"
+
+printf "Compiling: file $file\n"
+protoc --plugin=./node_modules/.bin/protoc-gen-ts_proto --ts_proto_out=./src/ $file
+printf "DONE\n"
+
+npm run format
```
Comment (kamil-stasiak) [inline compile_proto.sh:18]: enable trailing new line 🙃

File: src/components/ServerEvents.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { useServerSdk } from "./ServerSdkContext";
+import { ServerMessage } from "../protos/jellyfish/server_notifications";
+export const ServerEvents = ({ displayed }: { displayed: boolean }) => {
+  const [serverMessages, setServerMessages] = useState<ServerMessage[]>([]);
+  const { serverWebsocket } = useServerSdk();
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const handler = (event: any) => {
+    const uint8array = new Uint8Array(event.data);
+    console.log("called");
```
Comment (kamil-stasiak) [inline src/components/ServerEvents.tsx:12]: Remove console log

File: src/components/ServerSdkContext.tsx [hunk 3]
```diff
@@ -44,7 +44,21 @@ export const ServerSDKProvider = ({
...
   );
+  const httpServerUrl = signalingProtocol + "://" + signalingHost + signalingPath.replace("peer", "server");
+  const serverWebsocket = useMemo(() => (httpServerUrl ? new WebSocket(httpServerUrl) : null), [httpServerUrl]);
+  if (serverWebsocket) {
+    serverWebsocket.binaryType = "arraybuffer";
+    // create a new writer
+    const authRequestWriter = protobuf.Writer.create();
+    const subscribeRequestWriter = protobuf.Writer.create();
+    const buffer = authRequestWriter.uint32(58).fork().uint32(10).string("development").ldelim().finish();
+    const subscr = subscribeRequestWriter.uint32(66).fork().uint32(8).int32(1).ldelim().finish();
```
Comment (kamil-stasiak) [inline src/components/ServerSdkContext.tsx:55]: Can we use a more user-friendly version of this code? You mentioned that that is an inline version of some other method

File: src/utils/createMockAudio.tsx [hunk 4]
```diff
@@ -8,7 +8,7 @@ export const createMockAudio: () => {
   const gainNode = audioContext.createGain();
   gainNode.gain.value = 0.1;
   oscillator.connect(gainNode);
-  // gainNode.connect(audioContext.destination);
+  gainNode.connect(audioContext.destination);
```
Comment (kamil-stasiak) [inline src/utils/createMockAudio.tsx:11]: whats that?

**Review by kamil-stasiak — CHANGES_REQUESTED** [review]
Nice job! 
Only minor comments 👌

File: src/utils/createMockAudio.tsx (hunk 4 above)
Comment (Crackhoff) [reply src/utils/createMockAudio.tsx:11]: forgot to comment after testing, fixed

File: src/components/ServerSdkContext.tsx (hunk 3 above)
Comment (Crackhoff) [reply src/components/ServerSdkContext.tsx]: I like the crazy random numbers though :///
Fixed nonetheless

**Review by kamil-stasiak — APPROVED** [review]

**Pull request merged**